Automattic/jetpack
# jetpack: Add version to autoloader-suffix

#### Changes proposed in this Pull Request:

This allows for multiple release version of Jetpack to be installed side
by side, see #19472. It turns out that PHP namespace names can include
any non-ASCII Unicode character, so let's use "ⓥ" (U+24E5) to signal
that a version number is included.

Also, make sure this version is checked/updated by
tools/project-version.sh.

And for good measure, have the "Project structure" CI check require that
any project that sets "production-include" for vendor/autoload.php or
vendor/autoload_packages.php has autoloader-suffix set, and require that
any autoloader-suffix includes the version number indicator and is
reasonably long enough.

#### Jetpack product discussion


#19472

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Run `composer dump-autoloader` in Jetpack. See that the new suffix is included in the various files in `vendor/`.
* Does the plugin still work? No autoloader errors or anything?
* Use `tools/changelogger-release.sh` to prepare a Jetpack release. See that the version number in the autoloader suffix is updated.
* Change the version part of Jetpack's suffix. See that `tools/changelogger-validate-all.sh` catches it.
* Edit Jetpack's composer.json to remove the autoloader-suffix entirely. See that `.github/files/lint-project-structure.sh` catches it. Or submit it as a PR and see that the "Project structure" job catches it.
* Revert the part of this PR that changes Jetpack's composer.json. See that `.github/files/lint-project-structure.sh` catches it. Or submit it as a PR and see that the "Project structure" job catches it.
* Edit Jetpack's composer.json to set the autoloader-suffix to something short. See that `.github/files/lint-project-structure.sh` catches it. Or submit it as a PR and see that the "Project structure" job catches it.
